realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/SchemaTests.kt

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.realm.SyncTestUtils.Companion.createTestUser
+import io.realm.entities.StringOnly
+import junit.framework.Assert
+import junit.framework.TestCase
+import org.junit.*
+import org.junit.runner.RunWith
+import kotlin.test.assertFailsWith
+import kotlin.test.assertTrue
+
+@RunWith(AndroidJUnit4::class)
+class SchemaTests {
+    @get:Rule
+    val configFactory = TestSyncConfigurationFactory()
+
+    private var config: SyncConfiguration? = null
+    private var app: TestRealmApp? = null
+
+    @Before
+    fun setUp() {
+        app = TestRealmApp()
+        val user = createTestUser(app!!)
+        config = configFactory.createSyncConfigurationBuilder(user).build()
+    }
+
+    @After
+    fun tearDown() {
+        if (app != null) {
+            app!!.close()
+        }
+    }
+
+    @Test
+    fun instance() {
+        val realm = Realm.getInstance(config!!)
+        realm.use {
+            TestCase.assertFalse(realm.isClosed)
+        }
+        Assert.assertTrue(realm.isClosed)
+    }
+
+    @Test
+    fun createObject() {
+        val realm = Realm.getInstance(config!!)
+        realm.beginTransaction()
+        Assert.assertTrue(realm.schema.contains("StringOnly"))
+        val stringOnly = realm.createObject(StringOnly::class.java)
+        stringOnly.chars = "TEST"
+        realm.commitTransaction()
+        Assert.assertEquals(1, realm.where(StringOnly::class.java).count())
+        realm.close()
+    }
+
+    @Test
+    fun disallow_removeClass() {
+        // Init schema
+        Realm.getInstance(config!!).close()
+        val realm = DynamicRealm.getInstance(config!!)
+        val className = "StringOnly"
+        realm.beginTransaction()
+        Assert.assertTrue(realm.schema.contains(className))
+        // FIXME Why? We cannot update a sync schema "destructively"? -> Update Doc
+        //  Shouldn't this be UnsupportedOperation to be consistent with ex. addField, etc.
+        assertFailsWith<java.lang.IllegalArgumentException> {
```

Comment:
Yes, this should be `UnsupportedOperation` instead. Looks like a mistake here and we should fix it, but it is a breaking change, so will require a note in the CHANGELOG.md

Reply:
They should now be aligned. I have also added test cases for the other _destructive_  updates that could throw exceptions for this reason.